mlfinlab/ensemble/sb_bagging.py

```diff
@@ -0,0 +1,660 @@
+    .. [2] L. Breiman, "Bagging predictors", Machine Learning, 24(2), 123-140,
+           1996.
+    .. [3] T. Ho, "The random subspace method for constructing decision
+           forests", Pattern Analysis and Machine Intelligence, 20(8), 832-844,
+           1998.
+    .. [4] G. Louppe and P. Geurts, "Ensembles on Random Patches", Machine
+           Learning and Knowledge Discovery in Databases, 346-361, 2012.
+    """
+
+    def __init__(self,
+                 triple_barrier_events,
+                 price_bars,
+                 base_estimator=None,
+                 n_estimators=10,
+                 max_samples=1.0,
+                 max_features=1.0,
+                 bootstrap_features=False,
+                 oob_score=False,
+                 warm_start=False,
+                 n_jobs=None,
+                 random_state=None,
+                 verbose=0,
+                 ):
+        super().__init__(
+            triple_barrier_events=triple_barrier_events,
+            price_bars=price_bars,
+            base_estimator=base_estimator,
+            n_estimators=n_estimators,
+            max_samples=max_samples,
+            max_features=max_features,
+            bootstrap_features=bootstrap_features,
+            oob_score=oob_score,
+            warm_start=warm_start,
+            n_jobs=n_jobs,
+            random_state=random_state,
+            verbose=verbose)
+
+    def _validate_estimator(self):
+        """Check the estimator and set the base_estimator_ attribute."""
+        super(BaggingClassifier, self)._validate_estimator(
+            default=DecisionTreeClassifier())
+
+    def _set_oob_score(self, X, y):
+        n_samples = y.shape[0]
+        n_classes_ = self.n_classes_
+
+        predictions = np.zeros((n_samples, n_classes_))
+
+        for estimator, samples, features in zip(self.estimators_,
+                                                self.estimators_samples_(),
+                                                self.estimators_features_):
+            # Create mask for OOB samples
+            mask = ~indices_to_mask(samples, n_samples)
+
+            if hasattr(estimator, "predict_proba"):
+                predictions[mask, :] += estimator.predict_proba(
+                    (X[mask, :])[:, features])
+
+            else:
+                p = estimator.predict((X[mask, :])[:, features])
+                j = 0
+
+                for i in range(n_samples):
+                    if mask[i]:
+                        predictions[i, p[j]] += 1
+                        j += 1
+
+        if (predictions.sum(axis=1) == 0).any():
+            warn("Some inputs do not have OOB scores. "
+                 "This probably means too few estimators were used "
+                 "to compute any reliable oob estimates.")
+
+        oob_decision_function = (predictions /
+                                 predictions.sum(axis=1)[:, np.newaxis])
+        oob_score = accuracy_score(y, np.argmax(predictions, axis=1))
```

Comment:
You only take accuracy into account? I checked the parent class. I see the parent also only looks at accuracy. I would say lets add in F1, precision and so on but I don't think its a priority. 

Good to know that the oob_score in the parent is only on accuracy. I have been using it a lot.

Reply:
Yeah, as sklearn has only accuracy in oob, I've decided to preserve initial logic